src/type/unit/function/unit.js

```diff
@@ -43,7 +43,7 @@ export const createUnitFunction = /* #__PURE__ */ factory(name, dependencies, ({
       return Unit.parse(x, { allowNoUnits: true }) // a unit with value, like '5cm'
     },
 
-    'number | BigNumber | Fraction | Complex, string': function (value, unit) {
+    'number | BigNumber | Fraction | Complex, string | Unit': function (value, unit) {
       return new Unit(value, unit)
```

Comment:
If Unit stops accepting a string in the second constructor position here, as I'd recommend, there are two options:
1) Split this into two signatures, `number|BigNumber|Fraction|Complex, Unit` and `number|BigNumber|Fraction|Complex,string` with the second returning `new Unit(value, Unit.parse(name)`; or
2) Add an automatic type conversion `{from: 'string', to: 'Unit', convert: s => Unit.parse(s)}`

Conceptually I am fine with either but if examination of the code revealed other places where the conversion would simplify things I would lean toward option (2). Note in particular that (2) would immediately allow dropping the (single-argument) `string` signature from `math.unit` as well.

Reply:
Option 2 is pretty clean. It also simplifies Unit.to() and maybe others.

Reply:
Archiving this conversation for the future.